lib/modules/manager/kotlin-script/extract.ts

```diff
@@ -0,0 +1,35 @@
+import { regEx } from '../../../util/regex';
+import { MavenDatasource } from '../../datasource/maven';
+import type { PackageDependency, PackageFile } from '../types';
+
+const dependsOnRegex = regEx(
+  /@file\s*:\s*DependsOn\s*\(\s*(?<replaceString>"(?<groupId>.*):(?<artifactId>.*):(?<version>.*)")\s*\)/g
+);
+const repositoryRegex = regEx(
+  /@file\s*:\s*Repository\s*\(\s*"(?<repositoryName>.*)"\s*\)/g
+);
+
+export function extractPackageFile(fileContent: string): PackageFile | null {
+  const registryUrls: string[] = [...fileContent.matchAll(repositoryRegex)]
+    .map((match) => {
+      return match.groups?.repositoryName;
+    })
+    .filter((x): x is string => x !== null);
+
+  const deps: PackageDependency[] = [
+    ...fileContent.matchAll(dependsOnRegex),
+  ].map((match) => {
+    const dep: PackageDependency = {
+      currentValue: match.groups?.version,
+      depName: `${match.groups?.groupId}:${match.groups?.artifactId}`,
+      replaceString: match.groups?.replaceString,
+      datasource: MavenDatasource.id,
+      registryUrls: registryUrls.length > 0 ? registryUrls : null,
+    };
+    return dep;
+  });
```

Comment:
Can you add a filter here, e.g. to strip out any which are missing depName, maybe also currentValue? Up to you if you want to log a message about them or just drop them silently. Otherwise it can be very confusing later if there's a dep with empty depName.

Reply:
Yeah, good point! Will do.

Reply:
Instead if a filter, I used a more strict regex that won't catch dependencies with missing groupId, artifactId or version.